BoxContentSDK/BoxContentSDK/OAuth2/BOXAbstractSession.m

```diff
@@ -204,56 +204,38 @@ + (NSArray *)usersInKeychain
+        NSObject *generic = [keychainItemWrapper objectForKey:((__bridge id)kSecAttrGeneric)];
+        if ([generic isKindOfClass:[NSString class]]) {
+            NSString *keychainIdentifier = (NSString *)generic;
             
-            NSString *keychainIdentifier = (NSString *) object;
-            if ([keychainIdentifier hasPrefix:[self keychainIdentifierPrefix]])
-            {
-                NSString *userID = [self userIDFromKeychainIdentifier:keychainIdentifier];
-                if (userID.length > 0)
-                {
-                    NSString *jsonString = [[self keychainItemWrapperForUserWithID:userID] objectForKey:(__bridge id)kSecValueData];
+            // Only look at those that have our prefix
+            if ([keychainIdentifier hasPrefix:[self keychainIdentifierPrefix]]) {
+                
+                // Extract user-id from prefix, and credentials from keychain value data
+                NSString *userID = [self userIDFromKeychainIdentifier:(NSString *)keychainIdentifier];
+                if (userID.length > 0) {
+                    NSString *jsonString = [keychainItemWrapper objectForKey:(__bridge id)kSecValueData];
                     NSError *error = nil;
                     NSDictionary *dictionary = [NSJSONSerialization JSONObjectWithData:[jsonString dataUsingEncoding:NSUTF8StringEncoding] options:0 error:&error];
-                    NSString *userIDFromKeychain = [dictionary objectForKey:keychainUserIDKey];
-                    NSString *userNameFromKeychain = [dictionary objectForKey:keychainUserNameKey];
-                    NSString *userLoginFromKeychain = [dictionary objectForKey:keychainUserLoginKey];
-                    if ([userID isEqualToString:userIDFromKeychain]) {
-                        BOXUserMini *miniUser = [[BOXUserMini alloc] initWithUserID:userIDFromKeychain
-                                                                               name:userNameFromKeychain
-                                                                              login:userLoginFromKeychain];
-                        [users addObject:miniUser];
+                    if (error == nil) {
+                        NSString *userIDFromKeychain = [dictionary objectForKey:keychainUserIDKey];
+                        NSString *userNameFromKeychain = [dictionary objectForKey:keychainUserNameKey];
+                        NSString *userLoginFromKeychain = [dictionary objectForKey:keychainUserLoginKey];
+                        if ([userID isEqualToString:userIDFromKeychain]) {
+                            BOXUserMini *miniUser = [[BOXUserMini alloc] initWithUserID:userIDFromKeychain
+                                                                                   name:userNameFromKeychain
+                                                                                  login:userLoginFromKeychain];
+                            [users addObject:miniUser];
+                        }
                     }
```

Comment:
Anything we should do in the case of an error? returning nil ultimately for 'users' could be potentially misleading if there are indeed entries in the keychain...